### PR TITLE
Hubble: add option to filter for pods and services in any namespace

### PR DESCRIPTION
**Pull request by glrf**
Please ensure your pull request adheres to the following guidelines:

- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.

We now allow service and pod filters in the form of `/foo`, which will filter for any service/pod starting with `foo` in any namespace.

Related to: https://github.com/cilium/hubble/issues/359

### CODE REVIEW

**Review by chancez — APPROVED** [review]
Interesting idea, this seems reasonable to me.

File: pkg/hubble/filters/k8s.go [hunk 1]
```diff
@@ -37,10 +37,10 @@ func filterByNamespacedName(names []string, getName func(*v1.Event) (ns, name st
 	type nameFilter struct{ ns, prefix string }
 	nameFilters := make([]nameFilter, 0, len(names))
 	for _, name := range names {
-		ns, prefix := k8s.ParseNamespaceName(name)
-		if ns == "" && prefix == "" {
+		if name == "" {
 			return nil, fmt.Errorf("invalid filter, must be [namespace/][<name>], got %q", name)
```
Comment (rolinh) [inline pkg/hubble/filters/k8s.go]: The error message should probably be adjusted to account for the change.

**Review by rolinh — APPROVED** [review]
Nice! lgtm

**Review by kaworu — APPROVED** [review]
Thanks!

**Comment by glrf** [comment]
The CI failures seem unrelated and most likely was because this branch was >400 commits behind main/CI. Rebased on main. Could you please re-trigger tests? :)

**Comment by glrf** [comment]
Test failures seem like unrelated flakes:

* https://github.com/cilium/cilium/actions/runs/7038145700/job/19154490022 Fails during provisioning kind
* https://github.com/cilium/cilium/actions/runs/7038145505/job/19154442498 hit #24990 
* https://github.com/cilium/cilium/actions/runs/7038145539/job/19154443132 hit #27762

**Pull request merged**